GT Bees enhancements

Yefancy opened the pull request:
Some bee effects have been added for gt:
- `GT_ACCELERATE (Chrono)`, accelerate gt machines
- `GT_FIX(Treasure)`, fix the gt tool while players is in territory
- `GT_ENERGY (Quantum)`, power the gt machines around

(intensity is related to genes)

So far, only some simple features have been implemented.
I plan to do more stuff, if that's all right with you.
This PR could not be merged now, I will add some new features (related to forestry and Binnies).

huneau commented:
yeah nice feature

Yefancy commented on src/main/java/gregicadditions/integrations/bees/GTBees.java:
```diff
@@ -572,7 +491,7 @@ protected void registerMutations() {
 		String lowercaseName = this.toString().toLowerCase(Locale.ENGLISH);
 		String species = "species" + WordUtils.capitalize(lowercaseName);
 
-		String modId = Constants.MOD_ID;
+		String modId = Gregicality.MODID;
```
This place seems to be the code reference from the forestry. The original `Constants.MOD_ID` in here referred to jei (wrong package import?), causing all GA bees to be registered under `just enough items`.  An error will even be reported if jei is not installed.
Although I have modified it to `gregicality` here, it will cause the problem of bees missing in the save when the version is updated. This may need to be discussed.

huneau replied:
ok perfect

huneau commented on src/main/java/gregicadditions/integrations/bees/effects/GTBeesEffects.java:
```diff
@@ -0,0 +1,193 @@
+            return storedData;
+        }
+    },
+    GT_ENERGY {
```
`(long) (genome.getSpeed() * genome.getLifespan() * 10);` how many EU that can be ?

Yefancy replied:
The maximum is `70*1.7*10=1190 eu` per tick. The minimum is `10*0.3*10=30 eu` per tick.

huneau replied:
ok

huneau commented on src/main/java/gregicadditions/integrations/bees/effects/GTBeesEffects.java:
```diff
@@ -0,0 +1,193 @@
+            return new TickEffectData();
+        }
+    },
+    GT_FLUID {
```
what happens with a machine like Chemical reactor ? does it fill all fluid slot ?

Yefancy replied:
will fill all slots.

huneau replied:
ok

huneau commented on src/main/java/gregicadditions/machines/TileEntityWorldAccelerator.java:
```diff
@@ -91,7 +94,8 @@ protected long getMaxInputOutputAmperage() {
     public void update() {
         super.update();
-        if (!getWorld().isRemote) {
+        if (!getWorld().isRemote && lastTick != FMLCommonHandler.instance().getMinecraftServerInstance().getTickCounter()) {
```
what does it do ?

Yefancy replied:
yes, I don't think accelerators should be accelerated by possible other machines, such as the accelerator of other mods, that might cause a dead cycle. And the World accelerator is fast enough

huneau replied:
ok good idea

huneau commented on src/main/java/gregicadditions/GAConfig.java:
```diff
@@ -212,6 +212,9 @@
         public boolean ReactorRecipes = true;
 
+        @Config.Comment("Add Centrifuge recipes for the Combs")
+        public boolean CentrifugeRecipes = true;
```
there is already that 

![image](https://user-images.githubusercontent.com/9199240/102056991-8736ef80-3ded-11eb-8d58-0aeb4f328202.png)

huneau replied:
https://github.com/Gregicality/gregicality/blob/0.21/src/main/java/gregicadditions/GAConfig.java#L197

Yefancy replied:
commit: e7ec1f80614c863592c2ec7a4de3c8abfd5f1ea9

Yefancy commented:
![fluid bees](https://user-images.githubusercontent.com/18493855/102061586-9f7b2e80-3e2e-11eb-8779-e37c4ab0df9f.jpg)

huneau commented on src/main/java/gregicadditions/integrations/bees/GTBees.java:
```diff
@@ -11,20 +11,174 @@
+			registerMutation(OIL, ETHANOL, 5);
+		}
+	},
+	RADON(Materials.Radon, 0.05f) {
```
radon should be by product of nuclear material

Yefancy replied:
commit: e7ec1f80614c863592c2ec7a4de3c8abfd5f1ea9

huneau commented:
really really really nice job guys, I left 2 comments to fix

Yefancy commented:
Thank you :). I'll fix them later and adjust some details.

Yefancy commented:
finally finished stuff. mainly added features as follow:
- fluid bees mentioned above.
- added `MutationConditionFluid` that mutations require specfic fluids.
- added `MutationConditionMetaValueItem` that mutations require specfic GT items.
- added `Chrono`, accelerate gt machines.
- added `Treasure`, fix gt tools while players is in territory.
- added `Quantum`. produce energy.
- added `Mellifluous`. produce fluid.
- added a new Alveary `GT Alveary` used to output the fluid and energy produced by gt bees. Supporting the consumption of nuclear wastes can modify the environment (Mutation, Lifespan, and Production). Also input of the `MutationConditionMetaValueItem`.

Changes to the original work:
- Bees belonging to `branch: Metaliferis` with `effect: Mellifluous` can produce molten fluid.
- Bees belonging to `branch: Minerallis` require specific nuclear wastes to mutate.
- `naquadah (effect: Chrono)`  `platinum(effect: Quantum)`  `uranium(effect: Treasure)`  `iridium(speed: Fastest)`

![2020-12-18_01 08 55](https://user-images.githubusercontent.com/18493855/102522128-ceb0ca80-40d0-11eb-8b1e-15029aa6e10b.png)

huneau commented:
very nice, good job
